docs/TOKENS.md

```diff
@@ -13,133 +13,141 @@ Spoilers. My love for Astro has grown deeper.
+We will create a separate `handleGenerateToken` function to generate the token. This logic will allow us to create a token within the same repository without needing to use a `POST` request.
+
+To create the token, we use the `buildTokenWithUid` function from the `agora-token` package. We pass `expireTime` to the token and privilege expiration field.
+
+The privilege expiration time defines how long users will have their privileges within the channel. If the privilege time expires, the user can still listen to the call but can no longer publish. Once the token expires, they will lose access to the channel completely. In our case, we will set the token and privilege expiration to the same duration.
 
-The privilege expiration time defines how long the user will have their privileges within the channel and access to the channel in general. If the privilege time expires, the user can still listen to the call but can no longer publish. And once the token expires in general, they will lose access to the channel completely.
+Once a token is generated, we can send a successful response using our helper function.
 
 ```ts
 const APP_ID = import.meta.env.APP_ID;
 const APP_CERTIFICATE = import.meta.env.APP_CERTIFICATE;
 
-export async function GET({ params }: APIContext) {
+export async function POST({ request }: APIContext) {
     // ...
-    
-    const expireTime = 600;
-    const privilegeExpireTime = 600;
-    let token;
 
-    token = agoraToken.RtcTokenBuilder.buildTokenWithUid(APP_ID, APP_CERTIFICATE, params.channel, params.uid, role, expireTime, privilegeExpireTime);
+    const token = await handleGenerateToken({ channel, role: agoraRole, uid, expireTime })
 
-    return new Response(JSON.stringify({
-        rtcToken: token
-    }), { headers })
+    return sendSuccessfulResponse({ token })
+}
+
+export async function handleGenerateToken({ channel, role, uid, expireTime }: { channel: string, role: number, uid: string, expireTime: number }) {
+    return agoraToken.RtcTokenBuilder.buildTokenWithUid(APP_ID, APP_CERTIFICATE, channel, uid, role, expireTime, expireTime);
 }
 ```
 
 ## Run the Backend
-Run this token generator using `npm run dev`; now you can type arbitrary values into the URL to receive a token. For example, the URL `localhost:4321/rtc/test/publisher/1.json` means the user with uid `1` requests a token for the `test` channel and wants to publish and receive data. When you type this into the URL you should receive a response of: 
+Run this token generator using `npm run dev`. You can use Postman to send a `POST` request with a body containing `channel`, `role`, `uid`, and `expireTime`. If your `POST` request is missing information, you will receive a Bad Request response telling you which information you missed. If you send all the necessary information, you will receive a successful response that looks like this:
```

Comment:
Postman is a great tool for regular usage and testing but you should also consider adding a curl command example for users to copy paste/edit and test the endpoint quickly to see if their response matches the one below.

Reply:
I was thinking CURL might be a bit of a pain to use for cloud recording, so by them using this Postman for this, it would be consistent for all the guides.

Or do you think all the guides should have curl?

Reply:
I'm a fan of Postman, but it takes extra knowledge/effort to set up. I think while curl can look intimidating but it allows users to do a simple copy paste, update vars and hit enter to send it.

Reply:
Do you think we should include Postman at all then?